File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnTertiaryFixedVariantDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnTertiaryFixedVariantDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd40
            AppTheme.STANDARD -> R.color.cstd40
            AppTheme.BLUE -> R.color.cbd40
            AppTheme.GREEN -> R.color.cgd40
            AppTheme.RED -> R.color.crd40
            AppTheme.NOTHING -> R.color.cnod40
        }
    }
}

File: lib/layer2_core/dark/e_on_tertiary_fixed_variant_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnTertiaryFixedVariantDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd40;
      case 'standard': return cstd40;
      case 'blue': return cbd40;
      case 'green': return cgd40;
      case 'red': return crd40;
      case 'nothing': return cnod40;
      default: return cstd40;
    }
  }
}
